### ISSUE TITLE
Deep copy error on modules implementing MessagePassing

### ISSUE THREAD

**Issue by dtortorella** [rusty1s/pytorch_geometric#941]
## 🐛 Bug

Since pytorch_geometric version 1.4 modules derived from the class `MessagePassing` cannot be deep copied (no issues with older versions).

## To Reproduce
Just run the following lines (same issue occurring with any other layer derived from `MessagePassing`)

```
import copy
import torch_geometric
gconv = torch_geometric.nn.GCNConv(1,1)
copy.deepcopy(gconv)
```

which produce:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.6/copy.py", line 180, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/opt/conda/lib/python3.6/copy.py", line 280, in _reconstruct
    state = deepcopy(state, memo)
  File "/opt/conda/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/opt/conda/lib/python3.6/copy.py", line 240, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/opt/conda/lib/python3.6/copy.py", line 169, in deepcopy
    rv = reductor(4)
TypeError: can't pickle mappingproxy objects
```

## Expected behavior

Should have produced a new deep copy of the object instead.

## Environment
Nvidia DGX station

Virtual environment with:
* torch 1.4
* torch_geometric 1.4.1

## Additional context

A deep copy is necessary in order to copy models or parts of models whose weighs are later altered.

**Comment by rusty1s** [comment]
Thanks for reporting and sorry for the error :(
I fixed it in master, see [here](https://github.com/rusty1s/pytorch_geometric/commit/dca0df6ced8a79bb81f2db5a96b8b5c9d232919d).

**Comment by dtortorella** [comment]
Thanks!

**Comment by tjiagoM** [comment]
I'm having the same issue when saving models with `torch.save`. @rusty1s is there any plan to include this fix in a release soon? Just to know whether I can wait or clone the repo in master. Thanks!

**Comment by rusty1s** [comment]
I can deploy a bugfix release next week :)

**Comment by tjiagoM** [comment]
Awesome!! :D

**Comment by rusty1s** [comment]
Fixed with `1.4.2`.
